src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx

```diff
@@ -91,14 +136,12 @@ const FilterGroup: React.FC<FilterGroupProps> = ({
               Cancel
             </Button>
             <Button
-              onClick={() => {
-                editFilter("name", newFilterName, filter);
-                resetEditState();
-              }}
+              disabled={!isValid}
+              onClick={handleSubmit}
               size="xsmall"
               variant={Variant.PrimaryOutline}
             >
-              OK
+              APPLY
```

Comment:
[nit] The component cases the text for us so no need to caps lock here
```suggestion
              Apply
```